core/arch/arm/kernel/generic_boot.c

```diff
@@ -551,6 +554,44 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_secure_device_node(void *fdt)
+{
+	const uint32_t *p;
+	int offs;
+	int len;
+#if (TRACE_LEVEL >= TRACE_INFO)
+	char s[33];
+	int n = 0;
+	int m;
+#endif
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "secure-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(secure_device_id))
+		len = sizeof(secure_device_id);
+
+	memcpy(secure_device_id, p, len);
+	secure_device_id_len = len;
+
```

Comment:
You miss the point ;) Well, not entirely obviously. At the moment, there is nothing wrong with having two DTs, one for secure world and the other one for normal world. But, I think it should be our goal to try and unite them. That's also the direction taken by Peter Maydell in his kernel patch (introducing the `secure-` prefix). Therefore I'd rather have OP-TEE assume that the DT might be shared with the kernel, so if there is some information inside, that is for OP-TEE to consume and that should not leak to NW, then OP-TEE had better clear it.

Thoughts?